src/it/scala/net/snowflake/spark/snowflake/PushdownEnhancement02.scala

```diff
@@ -33,11 +33,13 @@ class PushdownEnhancement02 extends IntegrationSuiteBase {
   private var thisConnectorOptionsNoTable: Map[String, String] = Map()
   private val test_table_basic: String = s"test_basic_$randomSuffix"
   private val test_table_number = s"test_table_number_$randomSuffix"
+  private val test_table_rank = s"test_table_rank_$randomSuffix"
```

Comment:
To Reviewers: Below is the description for this PR:

Snowflake supports rank()/dense_rank() for SELECT query, but doesn’t support them in reading with COPY UNLOAD.
The pushdown analysis is performed in plan generation, it is impossible to know whether use_copy_unload is true or false, so they are pushdown anyway.
If use_copy_unload = true (default is false), the COPY UNLOAD fails. But, SC will retry without pushdown them. So, the job will succeed in the end.
PS: The COPY UNLOAD supported function list can be found at https://docs.snowflake.com/en/user-guide/data-load-transform.html#supported-functions

1. The original fix is done in https://github.com/snowflakedb/spark-snowflake/pull/255
2. The original fix is reverted before releasing SC 2.8.2 in https://github.com/snowflakedb/spark-snowflake/pull/295
3. Use “git cherry-pick <hash_code>” to bring back the fix. The production code is auto-merged. Do some manual merge for test case.
4. There are 2 comments in this PR.
    a. The first one is generated by  “git cherry-pick <hash_code>”.
    b. The second commits is to update the test cases. The original test cases only run for use_copy_unload = false. Update them to run for use_copy_unload = true. If use_copy_unload is true, the reading must be successful and the result to be the same.